airbyte-integrations/bases/debezium-v1-9-2/src/main/java/io/airbyte/integrations/debezium/internals/MySQLDateTimeConverter.java

```diff
@@ -50,13 +60,33 @@ private void registerDate(final RelationalColumn field, final ConverterRegistrat
         return DebeziumConverterUtils.convertDefaultValue(field);
       }
 
-      return switch (fieldType.toUpperCase(Locale.ROOT)) {
-        case "DATETIME" -> DateTimeConverter.convertToTimestamp(x);
-        case "DATE" -> DateTimeConverter.convertToDate(x);
-        case "TIME" -> DateTimeConverter.convertToTime(x);
-        case "TIMESTAMP" -> DateTimeConverter.convertToTimestampWithTimezone(x);
-        default -> throw new IllegalArgumentException("Unknown field type  " + fieldType.toUpperCase(Locale.ROOT));
-      };
+      switch (fieldType.toUpperCase(Locale.ROOT)) {
+        case "DATETIME":
+          if (x instanceof Long) {
+            if (getTimePrecision(field) <= 3) {
+              return DateTimeConverter.convertToTimestamp(Conversions.toInstantFromMillis((Long) x));
+            }
+            if (getTimePrecision(field) <= 6) {
+              return DateTimeConverter.convertToTimestamp(Conversions.toInstantFromMicros((Long) x));
+            }
+          }
+          return DateTimeConverter.convertToTimestamp(x);
+        case "DATE":
+          if (x instanceof Integer) {
+            return DateTimeConverter.convertToDate(LocalDate.ofEpochDay((Integer) x));
+          }
+          return DateTimeConverter.convertToDate(x);
+        case "TIME":
+          if (x instanceof Long) {
+            long l = Math.multiplyExact((Long) x, TimeUnit.MICROSECONDS.toNanos(1));
+            return DateTimeConverter.convertToTime(LocalTime.ofNanoOfDay(l));
+          }
+          return DateTimeConverter.convertToTime(x);
+        case "TIMESTAMP":
```

Comment:
just curious: did you test with different precisions? It seems kind of weird that debezium would have special handling for datetime but not timestamp (and that `timestamp` isn't actually in their table >.> )

Reply:
Yup. They have listed it in the [docs](https://debezium.io/documentation/reference/stable/connectors/mysql.html#mysql-temporal-types) as well

> Excluding the TIMESTAMP data type, MySQL temporal types depend on the value of the time.precision.mode connector configuration property.